spec/visual_regression_tests/all_components.rb

```diff
@@ -0,0 +1,41 @@
+require "rails_helper"
+require "percy"
+require "uri"
+
+describe "visual regression test runner Percy", type: :feature, js: true do
```

Comment:
Does this need to be part of Rspec? It doesn't really feel like this code is doing a test as there are no assertions. I was wondering if the use of RSpec was just due to ease of Capybara availability? If we could easily spin up the server and capybara in a rake task would that be preferable? (I don't mind helping on that)

Reply:
I haven't got any strong opinion on what triggers the visual regression test to run - a rake task is a good idea, and the only thing is that it shouldn't be run as part of the _default_ rake task.

I don't think that this should be a blocker. I'd like to get this in front of developers to find out whether this is actually useful or not before spending more time on it - and running it as a GitHub Action makes this an easy change to make at a later date.

If this this workflow turns out to be useful then I'm happy to spend time on making it run in a better, more correct way.

Reply:
Yup agree on the default rake task. We define those in the [Rakefile](https://github.com/alphagov/govuk_publishing_components/blob/dcda1a5ad918b436c454ff4c2837e77fea7cc58c/Rakefile#L360) so we shouldn't be at risk.

I can see your concern about this being a blocker. There are lots of ways to consider it. It feels like the ideal approach would be for this to run implicitly as part of the unit tests (maybe part of some sort of render assertion) which could be switched on and off with an env var. But a few things to think about.

I did have a tinker around to see if I could work out how this could be disambuated from RSpec and came up with putting it in a `bin` script. See what you think as to whether this seems a viable way to avoid this being tangled with tests: https://github.com/alphagov/govuk_publishing_components/commit/87bc88cd3308ecf1a1fc6bccdf2ecd75803e7469

Reply:
Sorry to wade in with an opinion. I can see both sides of the argument here - it does feel a bit polluty to have this part of rspec but as that's what came out of the box plus we don't yet know what problems are really perhaps we should look to get this merged and in use, and collect some data? We have Kev's mini-spike to fall back on if we need to which is great. Feels like the sooner FE devs can make use of this the better.

Reply:
Thanks for taking the time to have a deep dive into this Kevin - I really appreciate the time you've taken investigate the rake task and the rest of the pull request.

I've had a ponder on whether this should or shouldn't be a rake task - and despite my earlier thoughts, now I think that this is spec / test and should be run using RSpec. 

 - Though the assertion isn't in the file, there a visual "assertion" that runs using Percy - so it is a test and should be treated one
 - The answer to "Where's the visual regression test?" should be "In the folder with the other tests"
 - I think that the set up for running this in RSpec is a lot clearer to understand than in the script - there's a lot of things that comes for free with RSpec that need to be set up in the script.
 - Frontend developers need to be able to easily update this - and RSpec is a tool that we're familiar with. This means that this is less likely to just gather dust when something needs updating in the future.
 - Excluding it from the default set of tests is because it's expensive to run in terms of both time and money, not because it isn't a test

If this causes problems in the future, we can and should revisit this - but for now using RSpec is fine and shouldn't be a blocker for merging this in.

Reply:
Sure, thanks for your perspectives here and no problem for chiming in @chao-xian - it's great to have other insights.

I get what you mean with your points. I do actually agree that Percy and RSpec should be together. My concerns here are that we're somewhat using RSpec as a mechanism to run a script, rather than actually integrating Percy and RSpec or organising the file like a spec - so it might be better a separation. However you have both raised good points too.

I do appreciate that the reorganisation aspect is beyond the scope of this first pass, so the remaining part I'm keen to resolve is doing non standard RSpec usage so this runs in some places not others. I've tried to come up with a counter proposal for that by using tags: https://github.com/alphagov/govuk_publishing_components/commit/09f8596820df372a4a41dd250e4410bcdddfad87

How does that sound?